scipio/src/sys/uring.rs

```diff
@@ -769,10 +769,15 @@ impl Reactor {
         self.queue_standard_request(source, UringOpDescriptor::PollAdd(flags));
     }
 
-    pub(crate) fn write_dma(&self, source: &Source, buf: &DmaBuffer, pos: u64) {
-        //        let op = UringOpDescriptor::WriteFixed(buf.as_ptr() as *const u8, buf.len(), pos, buf.slabidx);
-        let op = UringOpDescriptor::WriteFixed(buf.as_ptr() as *const u8, buf.len(), pos, 0);
-        self.queue_storage_io_request(source, op);
+    pub(crate) fn write_dma(&self, source: &Source, pos: u64) {
+        match &*source.source_type() {
+            SourceType::Write(_, buf) => {
+                let op =
+                    UringOpDescriptor::WriteFixed(buf.as_ptr() as *const u8, buf.len(), pos, 0);
```

Comment:
```suggestion
                    UringOpDescriptor::WriteFixed(buf.as_ptr(), buf.len(), pos, 0);
```

I think this is a no-op cast?